Assets/Scripts/NPCs/StateMachine/States/Chase.cs

```diff
@@ -31,21 +39,23 @@ private void Chasing()
 
     private Vector3 ComputeChaseVelocity()
     {
-        Vector3 goalLocation;
-        
+        Vector3 targetLocation;
+
         //The NPC will predict the future position of the player (pursuit behavior)
-        if (_npcManager.GetDistanceWithPlayer() > 1)
+        if (_npcManager.GetDistanceWithPlayer() > NPCsGlobalVariables.ChasePlayerRange)
```

Comment:
The range is very small. It would be strange if the player goes around a tree or turn the corner of a building and the NPCs stop chasing him. 

If you take COD zombies, for example, the zombies always know where you are.

Reply:
I agree with Alessandro on this ^

Reply:
Sorry, that's not what i meant
I meant that the NPC should not go in a straight line if it can't see the player
Because a wall might be in the way

And in this case it should use the Astar to find the path to the player

Reply:
The problem is that all of the walls are in houses and the houses always have a width greater than 4, so this should never occur

Reply:
It can occur diagonaly

Reply:
the problem that you are talking about will be fixed with collision avoidance in #25